library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -282,53 +226,51 @@ impl<T, A: Allocator> VecDeque<T, A> {
 
     /// Copies a contiguous block of memory len long from src to dst
     #[inline]
-    unsafe fn copy_nonoverlapping(&self, dst: usize, src: usize, len: usize) {
+    unsafe fn copy_nonoverlapping(&mut self, src: usize, dst: usize, len: usize) {
```

Comment:
hmm, reversing the meaning of same-typed parameters to an unsafe method is scary to me, but I guess you changed it to match `ptr::copy_nonoverlapping` which is a net win so seems fine.

Reply:
It indeed is scary, and I triple checked all the uses of both `copy_nonoverlapping` and `copy` to make sure they're properly updated. Considering that I didn't see any errors after a few hours of fuzzing, I'm reasonably certain all the call sites are using the proper order.